packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -51,16 +75,97 @@ const RENDER_APIS: { [primitive in RenderPrimitive]: RenderPrimitiveDefinition }
     scopedFragId: { name: 'fid', alias: 'api_scoped_frag_id' },
     comment: { name: 'co', alias: 'api_comment' },
     sanitizeHtmlContent: { name: 'shc', alias: 'api_sanitize_html_content' },
+    setOwner: { name: 'so', alias: 'api_set_owner' },
 };
 
 interface Scope {
     parent: Scope | null;
     declaration: Set<string>;
 }
 
+function getStaticNodes(root: Root): Set<ChildNode> {
+    const staticNodes = new Set<ChildNode>();
+
+    function isStaticNode(node: BaseElement, parent: BaseParentNode): boolean {
+        let result = true;
+        const {
+            name: nodeName,
+            namespace = '',
+            attributes,
+            directives,
+            properties,
+            listeners,
+        } = node;
+
+        // Note: the trick is that the static nodes elm can not be removed by themself, they will always be removed
+        //       as parent removal.
+
+        result &&= !isIf(parent); // the parent node is an if, therefore this it is the same as an inline if
```

Comment:
Sure, the gist is that if this node is within a `template if:`, then it can't be static because it means that the diffing algo may remove the vnode.element directly, and at the point of removal, there's a 99% chance that the hoisted vnode does not contain the correct element to be removed.